### DISCUSSION TITLE
We need to stop the search function from searching the text in the document. It's bringing back too many results

### DISCUSSION THREAD

**Discussion by supreet-bajaj** [microsoft-search/pnp-modern-search#3516] [General]
Hi Team,

I have been using pnp search web part but it’s showing search results from inside documents as well when searching something in search text box.
I need to stop the search function from searching the text inside of the document as It's bringing back too many results.

can you please suggest how I can achieve that?

Thanks.
Preet

**Comment by supreet-bajaj** [comment]
Can someone please assist with this?

**Comment by kasperbolarsen** [comment]
Hi,
This question is not really related to PnP Modern Search as it is a very generic Search question. Perhaps you will get better answers using StackOverflow or similar?

**Comment by supreet-bajaj** [comment]
Hi Kasper,

I have been using pnp search box with it's pnp search results web part.

It's related with pnp modern search.

When I search something in search box, it searches inside of document as
well with document title.

Let me know if there is any confusion.

Thanks,
Preet

**Comment by kasperbolarsen** [comment]
No confusion at all :-) Through PnP Modern Search you are using either SharePoint Search or Microsoft Search and in both cases you will have to modify your query against the search engine in such a way that you get the result you are looking for.
Getting the search query right is very much a generic search related issue and in no way a PnP Modern Search specify question :-)

**Comment by supreet-bajaj** [comment]
Hi Kasper,

I am using SharePoint search to get documents from document library but
just want to check document title, if it matches with title then only it
should show document in search results web part .

At the moment, it's searchinh keyword inside document as well.

Here is the query.

{Searchterms} path: library url -contentclass STS_list_documentlibrary

Please help me, what change I need to make in this query?

Thanks, preet

**Reply by kasperbolarsen** [reply]
We are primarily helping with PnP Modern Search issues,  so I suggest you ask in a more  general forum
